modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/utils/logger/BallerinaLogHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.nativeimpl.utils.logger;
+
+import org.ballerinalang.bre.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for selecting the appropriate logger for a particular log statement.
+ *
+ * @since 0.89
+ */
+public class BallerinaLogHandler {
+    private static final Logger ballerinaRootLogger = LoggerFactory.getLogger("ballerina");
+
+    public static Logger getLogger(Context ctx) {
+        String packageDirPath =
+                ctx.getControlStackNew().getStack()[ctx.getControlStackNew().fp - 1].getCallableUnitInfo()
+                        .getPackageInfo().getPkgPath();
+
+        if (".".equals(packageDirPath) || packageDirPath == null) {
+            return ballerinaRootLogger;
+        } else {
+            // TODO: Refactor this later
+            return LoggerFactory.getLogger(ballerinaRootLogger.getName() + "." + packageDirPath);
```

Comment:
This won't be creating loggers. It looks it up in a map. A logger is created and added to the map only if the logger is not already there.